samcli/commands/pipeline/bootstrap/cli.py

```diff
@@ -87,6 +96,66 @@
     is_flag=True,
     help="Prompt to confirm if the resources are to be deployed.",
 )
+@click.option(
+    "--permissions-provider",
+    default=IAM,
+    required=False,
+    type=click.Choice(PERMISSIONS_PROVIDERS),
+    help="Choose a permissions provider to assume the pipeline execution role. Default is to use an IAM User.",
+)
+@click.option(
+    "--oidc-provider-url",
+    help="The URL of the OIDC provider.",
+    required=False,
+)
+@click.option("--oidc-client-id", help="The client ID configured to use with the OIDC provider.", required=False)
+@click.option(
+    "--github-org",
+    help="The GitHub organization that the repository belongs to. "
+    "If there is no organization enter the Username of the repository owner instead "
+    "Only used if using GitHub Actions OIDC for user permissions",
+    required=False,
+)
+@click.option(
+    "--github-repo",
+    help="The name of the GitHub Repository that deployments will occur from. "
+    "Only used if using GitHub Actions OIDC for permissions",
+    required=False,
+)
+@click.option(
+    "--deployment-branch",
+    help="The name of the branch that deployments will occur from. "
+    "Only used if using GitHub Actions OIDC for permissions",
+    required=False,
+)
+@click.option(
+    "--oidc-provider",
+    help="The name of the CI/CD system that will be used for OIDC permissions "
+    "we currently only support GitLab, GitHub, and Bitbucket",
+    type=click.Choice([GITHUB_ACTIONS, GITLAB, BITBUCKET]),
+    required=False,
+)
+@click.option(
+    "--gitlab-group",
```

Comment:
Some of these options don't play well with each other I'm assuming. There needs to be appropriate error messaging if they are specified together.

Reply:
Currently it takes all options then decides which option to use based on what `oidc-provider` is. I see your point is helpful. I'm thinking we can leave it to another PR for improvements which includes some refactoring.

Reply:
I learned from @xazhao that these new options, if not applicable, will just be ignored. I think it makes sense to handle them for now, as at least these different groups (GH Action, Gitlab and Bitbucket) won't conflict each other. But IMO there's a room to improve in terms of UX, to give customers clearer messages about what options they really need to supply.